lib/api/completeMultipartUpload.js

```diff
@@ -258,107 +259,121 @@ function completeMultipartUpload(authInfo, request, log, callback) {
             if (partLength > storedPartsAsObjects.length) {
                 return next(errors.InvalidPart, destBucket);
             }
-            // if extra parts, need to delete the data when done with completing
-            // mpu so extract the info to delete here
             const extraPartLocations = [];
-            if (partLength < storedPartsAsObjects.length) {
-                const extraParts = storedPartsAsObjects.slice(partLength);
-                extraParts.forEach(part => {
-                    const locations = part.locations;
-                    locations.forEach(location => {
-                        if (!location || typeof location !== 'object') {
-                            return;
-                        }
-                        extraPartLocations.push(location);
-                    });
-                });
-            }
+            const partUploadedCompleted = [];
 
             for (let i = 0; i < partLength; i++) {
                 const part = jsonList.Part[i];
                 // If the complete list of parts sent with
                 // the complete multipart upload request is not
-                // in numerical order
-                // return an error
+                // in ascending order return an error
                 const partNumber = Number.parseInt(part.PartNumber[0], 10);
-                if (partNumber !== i + 1) {
-                    // If not in order return InvalidPartOrder
-                    if (partNumber <= partLength) {
+                if (i > 0) {
+                    const previousPartNumber =
+                      Number.parseInt(jsonList.Part[i - 1].PartNumber[0], 10);
+                    if (partNumber <= previousPartNumber) {
```

Comment:
Hi @mikeys, 

There is an actual meaning to the array order.

Tested against AWS:

```
{
  "Part": [
    { "PartNumber": 1, "ETag": "..." },
    { "PartNumber": 2, "ETag": "..." },
    { "PartNumber": 3, "ETag": "..." }
  ]
}
```
returns no error.
However, 
```
{
  "Part": [
    { "PartNumber": 1, "ETag": "..." },
    { "PartNumber": 3, "ETag": "..." },
    { "PartNumber": 2, "ETag": "..." }
  ]
}
```
returns 400 - InvalidPartOrder error